tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -981,28 +981,37 @@ cc_library(
 )
 
 cc_library(
-    name = "gpu_compiler_impl",
-    srcs = [
+    name = "gpu_compiler",
+    deps = if_cuda_is_configured([
```

Comment:
Please pull in gpu_compiler.h and gpu_compiler.cc as hdrs and srcs here, respectively, plus the required deps. This way you only need to add what nvptx_compiler and amdgpu_compiler need in addition to the common things.

Reply:
@thomasjoerg `nvptx_compiler` and `amdgpu_compiler` are subclasses of `gpu_compiler`. doing so would cause trigger cyclical dependency.

Reply:
@thomasjoerg I've addressed your other feedback and submitted a new version. For this particular request (keep `gpu_compiler.*` in `gpu_compiler` target), actually it was my original intention but then I ran into bazel detecting cyclical dependency and failed. I'll give it another try but could you give additional hints on this?

Reply:
I think I've got a way to get around the issue. Will revise the PR.

Reply:
Where does the cyclic dependency get introduced? Both, nvptx_compiler and amdgpu_compiler would depend on gpu_compiler, but the opposite should not be true, is it? Maybe I'm missing something. Do you still have the Bazel error output around?

Reply:
let me try reconstruct them. i forfeited several difference approaches.

Reply:
@thomasjoerg I have to admit I'm not well versed in bazel. I finally came up with a version which builds.

Reply:
Thanks for giving it another try. I wonder whether we can further reduce usage of `if_xxxx_is_configured` in the BUILD file. I'll import this PR, take a look, and report back.